### PR TITLE
some improvements in sage/graphs/orientations.py

### PR DESCRIPTION
**Pull request by dcoudert**
We do some improvements in file `src/sage/graphs/orientations.py` and in particular in method `strong_orientations_iterator`.

The next step will be to move all methods related to orientations from `graph.py` to `orientations.py`.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/graphs/orientations.py [hunk 1]
```diff
@@ -25,17 +25,31 @@
...
-    Returns an iterator over all strong orientations of a graph `G`.
+    Return an iterator over all strong orientations of a graph `G`.
 
-    A strong orientation of a graph is an orientation of its edges such that
-    the obtained digraph is strongly connected (i.e. there exist a directed path
-    between each pair of vertices).
+    A strong orientation of a graph is an orientation of its edges such that the
+    obtained digraph is strongly connected (i.e. there exist a directed path
+    between each pair of vertices). According Robin's theorem (see the
+    :wikipedia:`Robbins_theorem`), the graphs that have strong orientations are
```
Comment (mkoeppe) [inline src/sage/graphs/orientations.py:51]: Robin's or Robbin's or Robbins'?

Comment (mkoeppe) [reply src/sage/graphs/orientations.py:51]: According *to*, I think

File: src/sage/graphs/orientations.py [hunk 2]
```diff
@@ -171,7 +178,6 @@ def strong_orientations_iterator(G):
         # launch the algorithm for enumeration of the solutions
         for sol in _strong_orientations_of_a_mixed_graph(Dg, V, treeEdges):
```
Comment (mkoeppe) [inline src/sage/graphs/orientations.py]: this can be replaced by `yield from ....`

**Review by mkoeppe — COMMENTED** [review]
Otherwise LGTM

**Comment by dimpase** [comment]
I am mostly away from the keyboard, until mid August, and then again till early September.

File: src/sage/graphs/orientations.py (hunk 1 above)
Comment (dcoudert) [reply src/sage/graphs/orientations.py:51]: Fixed.

File: src/sage/graphs/orientations.py (hunk 2 above)
Comment (dcoudert) [reply src/sage/graphs/orientations.py]: Indeed.

**Review by mkoeppe — APPROVED** [review]

**Comment by dcoudert** [comment]
Thank you.

**Pull request merged**